### ISSUE TITLE
Expandable and collapsible headings and a content overview

### ISSUE THREAD

**Issue by Alagts** [TYPO3-Documentation/render-guides#665]
**Overview:**

Each page should ideally have a small content overview (headlines of the page) representing the overall structure of the page. 
Every of these points/headlines should link to the actual article with corresponding headline on that page. In the content overview there should be a toggle to expand all existing sections again to revert all headline states again. At the beginning when accessing the page all sections should be expanded by default. 

**Main functionality:** 

Each headline should be collapsible. When a headline (for example H2) is collapsed all headlines in it that are smaller than the collapsed headline (for example H3) should be invisible until the documentation reader expands the same headline (H2) again. 

**Main benefits:**
- accessibility
- improves readability
- gives the opportunity to focus on the topics of interest
